### DISCUSSION TITLE
AI integration

### DISCUSSION THREAD

**Discussion by GerardPolloRebozado** [docmost/docmost#229] [Ideas]
Some useful feature cloud be:

- Rewrite the text in a more formal way
- Generate image for my notes
- Convert a writing into a draw.io schema ( saw a PR about its integration)

All this could be done with external AI like [openAI,](https://platform.openai.com/) [groq](https://console.groq.com/) or [stability ai](https://platform.stability.ai/) and also locally with [ollama ](https://ollama.com/) or even with webgpu and [webllm](https://github.com/mlc-ai/web-llm)

**Comment by delta-whiplash** [comment]
It could be such a nice add

**Reply by bolavefasfas** [reply]
it would be great

**Comment by WavedirectTel** [comment]
Add me to the list

**Comment by semiautomatix** [comment]
I use AI with Notion and it's preventing me from moving to any self hosted option.

**Comment by rshewade** [comment]
Is there a way I can use Docmost with N8N ?

**Reply by rafaribe** [reply]
Have you found a way?

**Comment by grimwiz** [comment]
I'd use it to pull different pieces of information from different pages into an answer
Like treating my docmost folder of knowledge as an LLM's RAG.

**Comment by ErbusMyth** [comment]
I think AI Is A Bit Complicated Compared To the specified usecase .

**Comment by WavedirectTel** [comment]
Yes I don't see a case use for this other than using it for helping to write documentation.  That being said adding a chatbot into the program with a local LLM isn't a big deal but would require GPU.  Therefore unless you just use some form of API communication to a 3rd party provider you could integrate something like Anything LLM into it and point to an outside LLM provider.  

At this point though as grimwiz said I think it really is better to use your own and use the docs as your RAG.  Have your own AI write the docs.

**Comment by physics515** [comment]
I would love to connect to my instance Ollama and use Docmost as RAG and be able to query all of my docs across spaces.

**Comment by DmitryKalibrov** [comment]
It's will be great

**Comment by rafaribe** [comment]
Would love to see this worked on, I would rather have Local LLM instead of OpenAI but I guess some abstraction can be built that takes care of that

**Comment by wertania** [comment]
It would be nice to have a feature like "re-write".
A user is typing some text with bad structure and without caring typos and format. -> "Re-Write" -> Well formatted and text and better sentences.

**Comment by Julrob199** [comment]
If I would not mind about privacy by giving my data to some AI, I could directly use Notion and their AI. And local LLMs are in many cases too GPU-demanding, I think most people dont use a server with GPU.

I dont say, that this feature is not useful and might help someone. But not highest priority in my opinion. There are still some basic feature lacking (tag support, ... and see other top voted ideas). Just my 50 cents.

**Comment by hendrikebbers** [comment]
We are a multilingual team, and the "translate to language X" feature in Notion is a killer feature for us.
I believe that a general AI integration based on the OpenAI API (many providers are compatible with that one) would be a really solid start. By doing so everybody could configure an API endpoint (OpenAI API or custom one) and an API token and Docmost would provide some nice AI based features like "rewrite the text", "create a summary of this page", "translate to language X". For all that csutom context description with placeholders for the text would be good enough as a start. We did something similar for slack and I'm willing to help if somebody points me to the correct touchpoints in docmost

**Reply by Julrob199** [reply]
Ahh i see, that makes sense!

**Reply by hendrikebbers** [reply]
@Philipinho if that makes sense to you I'm happy to create an issue with a detailed description and onces defined I assume a member of my team can work on it

**Comment by clucraft** [comment]
Add me to the list. Would like to see integration of OpenWebUI for self-hosted AI LLM's.

**Comment by hazhayder** [comment]
Any updates on this?

**Reply by WavedirectTel** [reply]
I just copy and paste what I need from AI into it.  It formats it similarly so usually comes in really clean.  Having some sort of clawdbot locally could help but that still is pretty risky these days.
